hack/generator/pkg/astmodel/allof_type.go

```diff
@@ -27,13 +27,15 @@ type AllOfType struct {
 	types TypeSet
 }
 
+var _ Type = &AllOfType{}
+
 // MakeAllOfType is a smart constructor for AllOfType,
 // maintaining the invariants. If only one unique type
 // is passed, then the result will be that type, not an AllOf.
 func MakeAllOfType(types ...Type) Type {
```

Comment:
minor: `NewAllOfType` if it returns a ptr I think... although this is a bit of a special case because the return type is not actually `AllOfType` so maybe it doesn't make sense?

Reply:
Yeah, it's a smart constructor so doesn't always. I might rename these to `Build*`